mesos/src/main/java/tachyon/mesos/TachyonFramework.java

```diff
@@ -0,0 +1,228 @@
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.mesos;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.mesos.Protos;
+import org.apache.mesos.Scheduler;
+import org.apache.mesos.SchedulerDriver;
+import org.apache.mesos.MesosSchedulerDriver;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.util.FormatUtils;
+import tachyon.util.io.PathUtils;
+
+public class TachyonFramework {
+  static class TachyonScheduler implements Scheduler {
+    private boolean mMasterLaunched = false;
+    private String mMasterHostname = "";
+
+    @Override
+    public void disconnected(SchedulerDriver driver) {
+      System.out.println("Disconnected from master.");
+    }
+
+    @Override
+    public void error(SchedulerDriver driver, String message) {
+      System.out.println("Error: " + message);
+    }
+
+    @Override
+    public void executorLost(SchedulerDriver driver, Protos.ExecutorID executorId,
+        Protos.SlaveID slaveId, int status) {
+      System.out.println("Executor " + executorId.getValue() + " was lost.");
+    }
+
+    @Override
+    public void frameworkMessage(SchedulerDriver driver, Protos.ExecutorID executorId,
+        Protos.SlaveID slaveId, byte[] data) {
+      System.out.println("Executor: " + executorId.getValue() + ", " + "Slave: "
+          + slaveId.getValue() + ", " + "Data: " + data + ".");
+    }
+
+    @Override
+    public void offerRescinded(SchedulerDriver driver, Protos.OfferID offerId) {
+      System.out.println("Offered " + offerId.getValue() + " rescinded.");
+    }
+
+    @Override
+    public void registered(SchedulerDriver driver, Protos.FrameworkID frameworkId,
+        Protos.MasterInfo masterInfo) {
+      System.out.println("Registered framework " + frameworkId.getValue() + " with master "
+          + masterInfo.getHostname() + ":" + masterInfo.getPort() + ".");
+    }
+
+    @Override
+    public void reregistered(SchedulerDriver driver, Protos.MasterInfo masterInfo) {
+      System.out.println("Registered framework with master " + masterInfo.getHostname() + ":"
+          + masterInfo.getPort() + ".");
+    }
+
+    @Override
+    public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offers) {
+      TachyonConf conf = new TachyonConf();
+      double masterCpu = conf.getInt(Constants.MASTER_MESOS_CPU);
+      double masterMem = conf.getBytes(Constants.MASTER_MESOS_MEM) / (1024 * 1024); // MBs
+      String tachyonHome = conf.get(Constants.TACHYON_HOME);
+      int launchedTasks = 0;
+
+      for (Protos.Offer offer : offers) {
+        Protos.Offer.Operation.Launch.Builder launch = Protos.Offer.Operation.Launch.newBuilder();
+        double offerCpu = 0;
+        double offerMem = 0;
+        for (Protos.Resource resource : offer.getResourcesList()) {
+          if (resource.getName().equals("cpus")) {
```

Comment:
NIT: better to define string constants on "cpus" and "mem" and use the string consts rather than having them inline. This is because you are using the string value "cpus" multiple times and referencing them in string consts help compilers catch misspelling.